internal/controllers/topology/cluster/reconcile_state_test.go

```diff
@@ -1801,10 +1789,31 @@ func TestReconcileMachineDeploymentMachineHealthCheck(t *testing.T) {
 			current: nil,
 			desired: []*scope.MachineDeploymentState{
 				newFakeMachineDeploymentTopologyState("md-1", infrastructureMachineTemplate, bootstrapTemplate,
-					mhcBuilder.Build()),
```

Comment:
Q: does it makes sense to add `.Clone` to our builders, so we can avoid duplication of common settings, but re-use by cloning (without affecting the original builder)

Reply:
That's a good idea. I'll create an issue for that and do a first run at it - then we can open help-wanted issues for some of the other ones.

Reply:
@killianmuldoon can we implement it for this PR so we can reference it as an example in the issue?

Reply:
I'll review after this change is in.

Reply:
I've added a 'Clone()' method for the builder now.